compiler/src/iree/compiler/Codegen/Common/TileSizeSelection.h

```diff
@@ -15,6 +15,75 @@ namespace mlir::iree_compiler {
 using SizesAndScalableFlags =
     std::pair<SmallVector<int64_t>, SmallVector<bool>>;
 
+using SizeAndScalableFlag = std::tuple<int64_t &, bool &>;
```

Comment:
I'm about the make a refactor on this part and found this PR.

Is there a reason not to define a struct like:

```
struct ScalableTileSize {
  int64_t tileSize;
  bool scalableFlag;
};
```

Having field name seems to be clearer and easier to access them?

Reply:
What we really need is a pair of references (rather than a plain `int64_t` and a `bool`) to elements of 2 vectors. That’s because we do this in KernelDispatch.cpp
```cpp
SmallVector<int64_t> commonVecTileSizes = parallelVecTileSizes;
SmallVector<bool> commonVecScalableTileFlags = parallelVecScalableTileSizes;
```

Since one of the vectors contains `bool`, things are not as straightforward.

Btw, this alias is not really used (sorry, I was meant to remove it). But I appreciate that this code is not pretty regardless.